crates/commitlog/src/lib.rs

```diff
@@ -45,6 +51,16 @@ pub struct Options {
     ///
     /// Default: 65,535
     pub max_records_in_commit: NonZeroU16,
+    /// Whenever at least this many bytes have been written to the currently
+    /// active segment, an entry is added to its offset index.
+    ///
+    /// Default: 4096
+    pub offset_index_interval_bytes: NonZeroU64,
+    /// If `true`, require that the segment must be synced to disk before an
+    /// index entry is added.
```

Comment:
```suggestion
    /// index entry is added.
    ///
    /// Setting this to `true` implies that the commitlog is synced to disk
    /// every `offset_index_interval_bytes`, regardless of other settings.
    /// 
    /// Setting it to `false` (the default) will update the index whenever 
    /// the commitlog is synced _and_ `offset_index_interval_bytes` have
    /// been written. This may mean that less entries are written to the index
    /// than strictly every `offset_index_interval_bytes`.
```

Reply:
``` Setting this to `true` implies that the commitlog is synced to disk
   every `offset_index_interval_bytes`, regardless of other settings.```

umm why offset index setting will make commitlog to fsync?

What I understood from the proposa isl, if this is `true`, commitlog offset index will only be written when commitlog syncs to disk, and it does not care about `offset_index_interval_bytes` setting.
So, this would be the case where less entry may written to the disk than every `offset_index_interval_bytes`.

If set to `false`,  index entries will be written every `offset_index_interval_bytes` without irrespective about commitlog fsync.

This is implemented as per my understanding only in the PR.

Reply:
```suggestion    
    /// index entry is added.
    ///
    /// Setting this to `false` (the default) will update the index every
    /// `offset_index_interval_bytes`, even if the commitlog wasn't synced.
    /// This means that the index could contain non-existent entries in the
    /// event of a crash.
    ///
    /// Setting it to `true` will update the index when the commitlog is synced,
    /// and `offset_index_interval_bytes` have been written.
    /// This means that the index could contain fewer index entries than
    //// strictly every `offset_index_interval_bytes`.

```

More like this?

Reply:
Yes, this looks accurate